File: P4/civitasruby/lib/civitas/casilla_calle.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas

  class Casilla_calle < Casilla

    attr_reader :titulo

    def initialize(titulo)
      super(titulo.nombre)
      @titulo = titulo
    end
    
    @override
    def to_s
      "#{super}  #{@titulo.to_s}"
    end

    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        super
        jugador = todos.at(actual)
        if(!@titulo.tiene_propietario)
          jugador.puede_comprar_casilla
        else
          @titulo.tramitar_alquiler(jugador)
        end
      end
    end

  end
  
end

File: P4/Civitas/src/civitas/CasillaCalle.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author salva
 */
public class CasillaCalle extends Casilla{
    private TituloPropiedad titulo;
    private float importe; // para hacer más sencillo el acceso al precio compra
    
    public CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.titulo = titulo;
        this.importe = titulo.getPrecioCompra();
    }
    
    public TituloPropiedad getTituloPropiedad(){
        return titulo;
    }
    
    @Override
    public String toString(){
        String info = super.toString();
        info += "\nPrecio: " + importe;
        if (titulo.tienePropietario()){
            info += "\nPropietario: " + titulo.getPropietario().getNombre();
            info += "\nNúmero de casas: " + titulo.getNumCasas();
            info += "\nNúmero de hoteles: " + titulo.getNumHoteles();
            info += "\nPrecio de alquiler: " + titulo.getPrecioAlquiler();
        } else{
            info += "\nLa casilla no tiene dueño";
        }
        return info;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)) {
        informe(actual, todos);
        Jugador jugador = todos.get(actual);
        if (!titulo.tienePropietario()) {
            jugador.puedeComprarCasilla();
        } else {
            titulo.tramitarAlquiler(jugador);
        }
    }
}
}
